src/common/datetime.cpp

```diff
@@ -1723,7 +1769,9 @@ wxDateTime& wxDateTime::SetToWeekDayInSameWeek(WeekDay weekday, WeekFlags flags)
 
     if ( flags == Default_First )
     {
-        flags = GetCountry() == USA ? Sunday_First : Monday_First;
+        WeekDay firstDay;
```

Comment:
Wouldn't it make sense to abstract this block in a function? E.g. Replace this entire `if` with

    UseEffectiveWeekDayFlags(flags);